### ISSUE TITLE
Accept header with no q value is null not 1

### ISSUE THREAD

**Issue by jchannon** [aspnet/AspNetCore#2902]
When inspecting an Accept header with that has a media type with no `q` value the `Quality` value is null.

*Request:*
<img width="822" alt="screen shot 2018-02-24 at 20 56 02" src="https://user-images.githubusercontent.com/105126/36635289-0dd5c414-19aa-11e8-8bbc-9ee63c901e75.png">

*Server:*
<img width="1054" alt="screen shot 2018-02-24 at 21 01 35" src="https://user-images.githubusercontent.com/105126/36635292-1451a2d6-19aa-11e8-8e77-f870004e585a.png">

In the spec it states https://tools.ietf.org/html/rfc7231#section-5.3.1 `If no "q" parameter is present, the default weight is 1.`

ASP.Net Core has made the `Quality` value on `MediaTypeHeaderValue` a nullable double and if no `q` is found mark it is null.

When processing a request based on an Accept header I have to execute the following code for it to be correct ie/ default it to 1 if null

<img width="764" alt="screen shot 2018-02-24 at 21 17 11" src="https://user-images.githubusercontent.com/105126/36635319-86a1fffc-19aa-11e8-85cd-58b527d84bec.png">

**Comment by markrendle** [comment]
I was digging through the code after you mentioned this on Twitter, and it does appear to do the right thing. Specifically, look here: https://github.com/aspnet/Mvc/blob/c27b07ef3f8b07dfe1d1de38a8d7bf2d6a9298f4/src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs#L410

So while the Accept headers in the Request are not right, the ObjectResultExecutor using the DefaultOutputFormatterSelector using AcceptHeaderParser using MediaType.CreateMediaTypeSegmentWithQuality does use a default Quality value of 1.0d.

Basically, the GetTypedHeaders().Accept code is providing the right values, since the Headers don't give a `q` value, but MVC's own formatter pipeline treats the absence of a `q` value the right way. If you're using or implementing an alternative formatter pipeline, you need to do likewise.

**Comment by jchannon** [comment]
I'm doing this in Botwin https://github.com/jchannon/Botwin which is built
on top of ASP.Net Core directly similar to other libraries.

My preference would be that ASP.Net Core issue the fix rather than MVC and
Botwin and other libs have to work around it.

**Comment by Eilon** [comment]
I believe the code in MVC and HttpAbstractions was all merged into just HttpAbstractions, so tagging it for that.

**Comment by Tratcher** [comment]
MediaTypeHeaderValue is a raw representation of the header, so indicating the presence or the absence of the q value is meaningfull.  Otherwise you would not be able to round trip the header, it'd always be outputting q's. Defaulting to 1 if q is absent is a higher level concern.

**Comment by anurse** [comment]
Given @Tratcher 's assessment (which I agree with), I'm closing this. The header value is designed specifically to represent the data contained in the header.
